auth/src/main/java/io/grpc/auth/OAuth2ChannelInterceptor.java

```diff
@@ -39,40 +39,43 @@
 import io.grpc.ClientInterceptors.ForwardingCall;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
+import io.grpc.Status;
 
+import java.io.IOException;
 import java.util.concurrent.Executor;
 
-import javax.inject.Provider;
-
-/** Client interceptor that authenticates all calls with OAuth2. */
+/**
+ * Client interceptor that authenticates all calls with an OAuth2 credential.
+ * Uses the new and simplified Google auth library:
+ * https://github.com/google/google-auth-library-java
+ */
 public class OAuth2ChannelInterceptor implements ClientInterceptor {
```

Comment:
Unlike the V1 credentials, the new Credentials base type is not OAuth2 specific, so it may be worth using a more general name, such as AuthIntercepter or GoogleAuthIntercepter.